### PR TITLE
add: contract and abi

### PR DESCRIPTION
**Pull request by juroberttyb**
This PR

- Add solidity code of dao contract
- Add abi of the dao contract
- Update deployment config json
- Add a few more deploy config on other networks

## Summary by CodeRabbit

- **New Features**
	- Introduced a Decentralized Autonomous Organization (DAO) smart contract to manage members, roles, and interactions within the DAO.
	- Added a README section outlining tasks, including a deploy script and unit tests.

### CODE REVIEW

**Review by juroberttyb — COMMENTED** [review]
Currently not adding event.

File: smart-contracts/dao.sol [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+
+contract Dao {
+    // dao member => role => rater => rate
+    mapping(address => mapping(string => mapping(address => uint8))) public rates;
```
Comment (jo-elimu) [inline smart-contracts/dao.sol:7]:
```suggestion
    mapping(address => mapping(string => mapping(address => uint8))) public ratings;
```

File: smart-contracts/dao.sol [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    // all members of the dao
+    address[] public members;
+    mapping(address => bool) public isMember;
+
+    // roles of a given dao member
+    string[] public roles;
+    mapping(string => bool) public hasRole;
```
Comment (jo-elimu) [inline smart-contracts/dao.sol:15]:
```suggestion
    mapping(string => bool) public roleExists;
```

**Review by jo-elimu — CHANGES_REQUESTED** [review]
Added suggestions

File: smart-contracts/dao.sol [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    constructor() {
+        members.push(msg.sender);
```
[the remark this answers is not in the document]
Comment (juroberttyb) [reply smart-contracts/dao.sol]: Constructer removed.

File: smart-contracts/dao.sol (hunk 2 above)
Comment (juroberttyb) [reply smart-contracts/dao.sol]: Updated as suggested.

**Pull request merged**